### DISCUSSION TITLE
Does this do real-time or is it delayed?

### DISCUSSION THREAD

**Discussion by The-Cricket1** [shner-elmo/TradingView-Screener#42] [Q&A] [answered]
I only ask because I did not see it explicitly state its real-time. So does this pull data in real-time or is there a delay.

**Comment by shner-elmo** [comment, marked as answer]
This library is just a wrapper on the TV API. It doesn't authenticate for you or anything like that.

And the way the API currently works is that if you don't pass the cookies you will always get delayed data.

Even when TradingView offers free real-time data for that specific ticker. For example for `AAPL` which is part of the `NASDAQ` exchange, you can get real-time data for free (on the website), but using this library if you don't pass cookies TradingView wont bother to give you real-time data, even if its free.

And if you did buy a data add-on (like for `OTC` symbols) then you definitely want to log in by passing your cookies.

In short, you really should pass your own cookies to log in/authenticate, so you get real-time data when possible.

**Reply by shner-elmo** [reply]
Have a look at [#28 ](https://github.com/shner-elmo/TradingView-Screener/issues/28#issuecomment-2093002556)
